File: test-suite/src/test/java/io/micronaut/docs/function/client/aws/MaxFunction.java
package io.micronaut.docs.function.client.aws;

import io.micronaut.function.FunctionBean;

import java.util.function.Supplier;

@FunctionBean("max")
public class MaxFunction implements Supplier<Integer> {

    private final MathService mathService;

    public MaxFunction(MathService mathService) {
        this.mathService = mathService;
    }

    @Override
    public Integer get() {
        return mathService.max();
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/client/aws/MaxFunction.kt
package io.micronaut.docs.function.client.aws

import io.micronaut.function.FunctionBean

import java.util.function.Supplier

@FunctionBean("max")
class MaxFunction(private val mathService: MathService) : Supplier<Int> {

    override fun get(): Int {
        return mathService.max()
    }
}
